src/geometry/ArborX_HyperBox.hpp

```diff
@@ -0,0 +1,116 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_HYPERBOX_HPP
+#define ARBORX_HYPERBOX_HPP
+
+// FIXME_CXX17: remove first clause and update the comment once we require C++17
+#if (__cplusplus < 201703L) || (KOKKOS_VERSION < 30699)
```

Comment:
Why 3.6.99? `Kokkos` would require `C++17` for `KOKKOS_VERSION >= 3.7.99` (one the version update happened).

Reply:
There is no reason not to allow it working with Kokkos 3.7.

Reply:
Which features are you using here that require a newer `Kokkos` version than the minimum `ArborX` requires?

Reply:
New geometry classes don't have volatile overloads for the reduction, which requires 3.7.